talisker/gunicorn.py

```diff
@@ -275,13 +274,9 @@ def load_config(self):
                 extra={'logger_class': self.cfg.logger_class})
         # Use pip to find out if prometheus_client is available, as
         # importing it here would break multiprocess metrics
-        if (pkg_is_installed('prometheus-client') and
-                (self.cfg.workers or 1) > 1):
-            if 'prometheus_multiproc_dir' not in os.environ:
-                logger.info('running in multiprocess mode but '
-                            '`prometheus_multiproc_dir` envvar not set')
-                tmpdir = tempfile.mkdtemp()
-                os.environ['prometheus_multiproc_dir'] = tmpdir
-
-            logger.info('using `%s` for multiprocess prometheus metrics',
-                        os.environ['prometheus_multiproc_dir'])
+        dir = os.environ.get('prometheus_multiproc_dir')
+        if pkg_is_installed('prometheus-client') and dir is not None:
```

Comment:
Can `dir` be `None` here? What if it's `None`?

Reply:
right, it shouldn't be None ever, but defensively, it case it is None for some reason, don't log as the message as it's not true.